letsencrypt-auto-source/letsencrypt-auto.template

```diff
@@ -296,11 +307,21 @@ elif [ -f /etc/mageia-release ]; then
   }
   BOOTSTRAP_VERSION="BootstrapMageiaCommon $BOOTSTRAP_MAGEIA_COMMON_VERSION"
 elif [ -f /etc/redhat-release ]; then
-  Bootstrap() {
-    BootstrapMessage "RedHat-based OSes"
-    BootstrapRpmCommon
-  }
-  BOOTSTRAP_VERSION="BootstrapRpmCommon $BOOTSTRAP_RPM_COMMON_VERSION"
+  DeterminePythonVersion "NOCRASH"
+  if [ "$PYVER" -eq 26 ]; then
+    Bootstrap() {
+      BootstrapMessage "RedHat-based OSes that will use Python3"
+      BootstrapRpmPython3
+    }
+    export LE_PYTHON=python3
```

Comment:
Ok, changed some things around here and also inside `DeterminePythonVersion`. These are the cases I considered:

- their python as specified by `LE_PYTHON` doesn't exist 
- they didn't set le_python and they have no python installed
- le_python is 2.6
- le_python is 3.x
- le_python is 2.7

I can go through why I think this is the right decision for each of them, just not right now.

Reply:
#### LE_PYTHON affecting bootstrap selection

I'm unsure if we want to be taking `LE_PYTHON` set by the user into account when determining the bootstrap script to use.

My main reason for thinking this is the bootstrap script changing based on the user's environment can lead to some strange/obnoxious behavior. For example, you're running certbot-auto in cron as root and you get root email that says that certbot-auto needs to be run interactively to get permission to install new packages. You log in as your normal user and run certbot-auto. If you have `LE_PYTHON` set in one environment and not the other, a different bootstrap script may be selected in the two environments. This is similar to the issues we were having when there wasn't one Certbot install location on the system and it depended on the user's home directory.

If `LE_PYTHON` is set for the non-root user, it may not rebootstrap despite certbot-auto saying you have to. If `LE_PYTHON` is set for the root user, you may rebootstrap when running as a non-root user, but when certbot-auto is run again from cron, a different bootstrap script is selected so certbot-auto thinks you have to rebootstrap again!

If we do this, I think the only downside is we may install packages we may end up not needing, but doing this for the very few users who have been setting `LE_PYTHON` seems like it might be less harmful, confusing, and complex than the above behavior. If users know nothing else needs to be installed, there's a `--no-bootstrap` flag.

#### Overwriting LE_PYTHON

When setting `LE_PYTHON` here though, I think we should respect the value the user set it to and not just overwrite it. Setting `LE_PYTHON=python2.6` and having it suddenly run in Python 3 is a little confusing and unexpected. I think we should only set this value to `python3` if it wasn't set by the user.

My thought here is in the next release, you'll still be able to force Certbot to use Python 2.6 if you set `LE_PYTHON`. In the next release, we'll also start printing deprecation warnings if you're running in Python 2.6. In a subsequent release, we'll update `DeterminePythonVersion` to complain about your entombed Python version if it's less than 27 which should force you to stop using 2.6 entirely.